spec/system/candidate_interface/entering_details/candidate_choosing_visa_or_immigration_status_non_eu_spec.rb

```diff
@@ -0,0 +1,106 @@
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Student visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Student visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Graduate visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Graduate visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Skilled Worker visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Skilled Worker visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Dependent on partner’s or parent’s visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Dependent on partner’s or parent’s visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Family visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Family visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('British National (Overseas) visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('British National (Overseas) visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('UK Ancestry visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('UK Ancestry visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('High Potential Individual visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('High Potential Individual visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Youth Mobility Scheme')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Youth Mobility Scheme')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('India Young Professionals Scheme visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('India Young Professionals Scheme visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Ukraine Family Scheme or Ukraine Sponsorship Scheme visa')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Ukraine Family Scheme or Ukraine Sponsorship Scheme visa')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Afghan citizens resettlement scheme or Afghan Relocations and Assistance Policy')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Afghan citizens resettlement scheme or Afghan Relocations and Assistance Policy')
+
+    when_click_change_immigration_status
+    and_i_choose_the_visa('Refugee Status')
+    and_i_click_save_and_continue
+    then_i_should_see_the_correct_visa_in_the_summary('Refugee Status')
```

Comment:
I'm not sure if testing all the values should be done in a system spec.

Reply:
yeah, I wasn't keen either but I couldn't think of a better way to properly test it. Do you have any suggestions?

Reply:
You already wrote the specs, so might as well leave it now. I see system specs as a tool for testing interactions, not the data. And if you add another `immigration_status` value in the model, this spec will not fail so the dev might forget to add the scenario.